wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -2076,6 +2233,178 @@ private function translate_regexp_functions( $token ) {
 		return true;
 	}
 
+	/**
+	 * Detect GROUP BY.
+	 *
+	 * @param WP_SQLite_Token $token The token to translate.
+	 *
+	 * @return bool
+	 */
+	private function capture_group_by( $token ) {
+		if (
+			! $token->matches(
+				WP_SQLite_Token::TYPE_KEYWORD,
+				WP_SQLite_Token::FLAG_KEYWORD_RESERVED,
+				array( 'GROUP' )
+			)
+		) {
+			return false;
+		}
+		$next = $this->rewriter->peek_nth( 2 )->value;
+		if ( 'BY' !== strtoupper( $next ) ) {
+			return false;
+		}
+
+		$this->has_group_by = true;
+
+		return false;
+	}
+
+	/**
+	 * Translate WHERE something HAVING something to WHERE something AND something.
+	 *
+	 * @param WP_SQLite_Token $token The token to translate.
+	 *
+	 * @return bool
+	 */
+	private function translate_ungrouped_having( $token ) {
+		if (
+			! $token->matches(
+				WP_SQLite_Token::TYPE_KEYWORD,
+				WP_SQLite_Token::FLAG_KEYWORD_RESERVED,
+				array( 'HAVING' )
+			)
+		) {
+			return false;
+		}
+		if ( $this->has_group_by ) {
+			return false;
+		}
+		$this->rewriter->skip();
+		$this->rewriter->add( new WP_SQLite_Token( 'AND', WP_SQLite_Token::TYPE_KEYWORD ) );
+
+		return true;
+	}
+
+	/**
+	 * Rewrite LIKE '\_whatever' as LIKE '\_whatever' ESCAPE '\' .
+	 *
+	 * We look for keyword LIKE. On seeing it we set a flag.
+	 * If the flag is set, we emit ESCAPE '\' before the next keyword.
+	 *
+	 * @param WP_SQLite_Token $token The token to translate.
+	 *
+	 * @return bool
+	 */
+	private function translate_like_escape( $token ) {
+
+		if ( 0 === $this->like_expression_nesting ) {
+			$match = $token->matches( WP_SQLite_Token::TYPE_KEYWORD, null, array( 'LIKE' ) );
```

Comment:
A different variable name would make the logic clearer:
```suggestion
			$is_like_keyword = $token->matches( WP_SQLite_Token::TYPE_KEYWORD, null, array( 'LIKE' ) );
```